### PR TITLE
[PYTHON] Improve API with compact functions and Python wrappers

### PR DESCRIPTION
**Pull request by jiwaszki**
### Details:
 - Added compact function to compile models from single path to the file.
 - Improve Tensor API: removed call to module/interpreter during initialization, move convert to c-style array to pure C++ side.
 - Improve Python wrappers for InferRequest, ExecutableNetwork, AsyncInferQueue and Core. Instead of injecting functions, all of the advanced methods are wrapped on Python side via proper inheritance.
 - Removed unused classes/functions and cleaned up tests.

### Tickets:
 - *66277*

### CODE REVIEW

File: runtime/bindings/python/tests/test_inference_engine/test_core.py [hunk 1]
```diff
@@ -119,15 +105,15 @@ def test_read_model_from_onnx_as_path():
...
-@pytest.mark.xfail("68212")
-def test_read_net_from_buffer():
-    core = Core()
-    with open(test_net_bin, "rb") as f:
-        bin = f.read()
-    with open(model_path()[0], "rb") as f:
-        xml = f.read()
-    func = core.read_model(model=xml, weights=bin)
-    assert isinstance(func, IENetwork)
+# @pytest.mark.xfail("68212")
```
Comment (akuporos) [inline runtime/bindings/python/tests/test_inference_engine/test_core.py:108]: Why have you commented this test?

Comment (jiwaszki) [reply runtime/bindings/python/tests/test_inference_engine/test_core.py:108]: IENetwork is not existing now. Test will be enabled again and re-written with ticket 68212.

File: runtime/bindings/python/src/pyopenvino/core/common.hpp [hunk 2]
```diff
@@ -22,44 +24,23 @@ namespace py = pybind11;
...
 
     const Containers::TensorIndexMap cast_to_tensor_index_map(const py::dict& inputs);
 
-    const ov::runtime::Tensor& cast_to_tensor(const py::handle& tensor);
-
-    void blob_from_numpy(const py::handle& _arr, InferenceEngine::Blob::Ptr &blob);
+    PyObject* parse_parameter(const InferenceEngine::Parameter& param);
 
-    void set_request_blobs(InferenceEngine::InferRequest& request, const py::dict& dictonary);
+    PyObject* parse_parameter(const InferenceEngine::Parameter& param);
```
Comment (akuporos) [inline runtime/bindings/python/src/pyopenvino/core/common.hpp:43]: doesn't it duplicate line 41?

File: runtime/bindings/python/src/openvino/ie_api.py [hunk 3]
```diff
@@ -3,126 +3,109 @@
...
-# flake8: noqa: D102
 def tensor_from_file(path: str) -> Tensor:
-    """The data will be read with dtype of unit8"""
+    """Create Tensor from file. Data will be read with dtype of unit8."""
     return Tensor(np.fromfile(path, dtype=np.uint8))
+
+
+def normalize_inputs(py_dict: dict, py_types: dict) -> dict:
+    """Normalize a dictionary of inputs to Tensors."""
+    for k, v in py_dict.items():
```
Comment (akuporos) [inline runtime/bindings/python/src/openvino/ie_api.py:25]: may be rename ```k``` and ```v``` them to ```key``` and ```tensor``` to make it more verbose?

File: runtime/bindings/python/src/pyopenvino/core/executable_network.cpp [hunk 4]
```diff
@@ -20,28 +20,29 @@ void regclass_ExecutableNetwork(py::module m) {
...
     cls.def("create_infer_request", [](ov::runtime::ExecutableNetwork& self) {
         return InferRequestWrapper(self.create_infer_request(), self.inputs(), self.outputs());
     });
 
     cls.def(
-        "_infer_new_request",
+        "infer_new_request",
         [](ov::runtime::ExecutableNetwork& self, const py::dict& inputs) {
             auto request = self.create_infer_request();
             const auto key = inputs.begin()->first;
```
Comment (akuporos) [inline runtime/bindings/python/src/pyopenvino/core/executable_network.cpp:36]: looks like it isn't need anymore

File: runtime/bindings/python/src/pyopenvino/core/infer_request.cpp [hunk 5]
```diff
@@ -51,19 +57,18 @@ void regclass_InferRequest(py::module m) {
...
-                    for (auto&& input : inputs_map) {
-                        self._request.set_input_tensor(input.first, input.second);
+                for (auto&& input : inputs) {
+                    if (py::isinstance<py::str>(input.first)) {
+                        self._request.set_tensor(input.first.cast<std::string>(), Common::cast_to_tensor(input.second));
+                    } else if (py::isinstance<py::int_>(input.first)) {
+                        self._request.set_input_tensor(input.first.cast<size_t>(),
+                                                       Common::cast_to_tensor(input.second));
+                    } else {
+                        throw ov::Exception("Incompatible key type for tensor!");
```
Comment (akuporos) [inline runtime/bindings/python/src/pyopenvino/core/infer_request.cpp:71]: doesn't it double-check?
you checked whether ```key``` is ```int``` or ```string``` in the ```ie_api.py```

File: runtime/bindings/python/tests/test_inference_engine/test_infer_request.py [hunk 6]
```diff
@@ -186,9 +173,9 @@ def test_infer_mixed_keys(device):
     tensor2 = Tensor(data2)
 
     request = exec_net.create_infer_request()
-    with pytest.raises(TypeError) as e:
+    with pytest.raises(KeyError) as e:
```
Comment (akuporos) [inline runtime/bindings/python/tests/test_inference_engine/test_infer_request.py:176]: this test was about mixed keys, so as user now can pass mixed input dict check that it works

File: runtime/bindings/python/src/openvino/ie_api.py (hunk 3 above)
Comment (jiwaszki) [reply runtime/bindings/python/src/openvino/ie_api.py:25]: Rename to `val`

File: runtime/bindings/python/src/pyopenvino/core/infer_request.cpp (hunk 5 above)
Comment (jiwaszki) [reply runtime/bindings/python/src/pyopenvino/core/infer_request.cpp:71]: Discussed offline.

File: runtime/bindings/python/tests/test_inference_engine/test_core.py (hunk 1 above)
Comment (akuporos) [reply runtime/bindings/python/tests/test_inference_engine/test_core.py:108]: PR: https://github.com/openvinotoolkit/openvino/pull/8593

File: runtime/bindings/python/src/pyopenvino/core/infer_request.cpp (hunk 5 above)
Comment (akuporos) [inline runtime/bindings/python/src/pyopenvino/core/infer_request.cpp:71]: also py::type_error

File: runtime/bindings/python/src/pyopenvino/core/infer_request.cpp [hunk 7]
```diff
@@ -80,19 +85,18 @@ void regclass_InferRequest(py::module m) {
...
```
Comment (akuporos) [inline runtime/bindings/python/src/pyopenvino/core/infer_request.cpp:99]: also py::type_error

File: runtime/bindings/python/src/pyopenvino/core/infer_request.cpp (hunk 5 above)
Comment (jiwaszki) [reply runtime/bindings/python/src/pyopenvino/core/infer_request.cpp]: Fixed it with one commit. There can be common function for all classes that set tensors.

**Review by akuporos — APPROVED** [review]

**Pull request merged**